### ISSUE TITLE
How are the Distance Computation Tools used correctly?

### ISSUE THREAD

**Issue by KLC5CR6K** [CloudCompare/CloudComPy#19]
**How are the Distance Computation Tools (computeCloud2MeshDistances) used correctly?**

Previous Steps:
- [Docker environment](https://github.com/CloudCompare/CloudComPy/blob/master/doc/UseDockerLinuxConda.md) to use CloudComPy on Linux with miniconda
- Tested with version 8-cloudcompy-conda39-linux64-20211018

Used script:
```
    import cloudComPy as cc
    cc.initCC()

    cloud = cc.loadPointCloud( "cloud.asc" )
    ref_mesh = cc.loadMesh( "mesh.obj" )

    ret_cloud = cc.ccPointCloud()

    para = cc.Cloud2MeshDistancesComputationParams()
    para.flipNormals = False
    para.CPSet = ret_cloud

    cc.DistanceComputationTools.computeCloud2MeshDistances(cloud, ref_mesh, para)

    cc.SavePointCloud(ret_cloud, "out.asc")
```

I get the following error:
```
    TypeError: No to_python (by-value) converter found for C++ type: CCCoreLib::PointCloud*
```

Documentary read:
- https://www.simulation.openfields.fr/documentation/CloudComPy/html/DistanceComputationTools.html#cloudComPy.DistanceComputationTools.computeCloud2MeshDistances
- https://tmontaigu.github.io/CloudCompare-PythonPlugin/cpp/CCCoreLib/tools.html#_CPPv4N9CCCoreLib24DistanceComputationTools36Cloud2MeshDistancesComputationParams5CPSetE

**Comment by dgirardeau** [comment]
I see it's used in at least 2 tests? (009 and 022)

**Comment by KLC5CR6K** [comment]
Hi.

There is only one test. Sorry, what does 002 and 022 mean?

The problem specifically with the [dockerfile ](https://github.com/CloudCompare/CloudComPy/blob/dd08a6a33eab6e46ac4e0319caf172c691a787c6/doc/UseDockerLinuxConda.md)(commit dd08a6a). The binaries from 8-cloudcompy-conda39-linux64-20211018-tgz are used in the Dockerfile.

**Comment by dgirardeau** [comment]
Since you were asking about the correct way to use the computeCloud2MeshDistances function, I thought looking at the code of the tests that use this function would help:
https://github.com/CloudCompare/CloudComPy/blob/master/tests/test009.py
https://github.com/CloudCompare/CloudComPy/blob/master/tests/test022.py

**Comment by prascle** [comment]
Thank you @dgirardeau  for your help,
I still need to improve the documentation, and maybe the best way is to provide a kind of cross reference table giving a test script for each method, or write a lot of simple use case scripts with some explanations.

Here, para.CPSet is of type ReferenceCloud 
`para.CPSet = cc.ReferenceCloud(ret_cloud)`
But I have no  example of script using this CPSet. 
In the CloudCompare GUI, it is used internally in menus like _tools/Sand Box .../Compute best registration RMX Matrix_, but the user does not need to manipulate this parameter...

**Comment by dgirardeau** [comment]
Yes, the 'CPSet' (Closest points set) is only used internally by the ICP registration algorithm.
